src/components/playback/playbackmanager.js

```diff
@@ -261,18 +261,109 @@ define(['events', 'datetime', 'appSettings', 'itemHelper', 'pluginManager', 'pla
+                    Url: t.Url,
+                    MediaType: 'Video',
+                    Type: 'Trailer',
+                    ServerId: apiClient.serverId()
+                };
+            });
+        }
+
+        function getAllTrailers(item, remoteTrailers) {
+            if (item.LocalTrailerCount) {
+                return apiClient.getLocalTrailers(apiClient.getCurrentUserId(), item.Id).then(function (trailers) {
+                    trailers.push.apply(trailers, remoteTrailers);
+                    return trailers;
+                }).catch(function (err) {
+                    console.error(err);
+                    return remoteTrailers;
+                });
+            } else {
+                return Promise.resolve(remoteTrailers);
+            }
+        }
+
+        return apiClient.getItems(apiClient.getCurrentUserId(), fetchOptions).then(function (unwatchedMovies) {
+            // ensures that there are never fewer unwatched movies than there should be trailers
+            if (unwatchedMovies.Items.length < trailerCount) {
+                trailerCount = unwatchedMovies.Items.length;
+            }
+
+            var indexHistory = [];
+            var promises = [];
+
+            for (var i = 0; i < trailerCount && indexHistory.length < unwatchedMovies.Items.length; i++) {
+                // get random movies that hasn't been selected
+                var index = Math.floor(Math.random() * unwatchedMovies.Items.length);
+                while (indexHistory.includes(index)) {
+                    index = Math.floor(Math.random() * unwatchedMovies.Items.length);
+                }
+                indexHistory.push(index);
+                var randomMovie = unwatchedMovies.Items[index];
 
-            return result;
+                var remoteTrailers = getRemoteTrailers(randomMovie);
+
+                // don't count the movie to the trailer count if it has no trailers
+                if (!randomMovie.LocalTrailerCount && !remoteTrailers.length) {
+                    i--;
+                    continue;
+                }
+
+                promises.push(getAllTrailers(randomMovie, remoteTrailers));
+            }
+
+            return Promise.all(promises);
+
+        }).then(function (movieTrailers) {
+            var randomTrailers = [];
+
+            for (var i = 0; i < movieTrailers.length; i++) {
+                var trailers = movieTrailers[i];
+
+                // in case of error with local trailers and empty remote trailers
+                if (!trailers.length) {
+                    continue;
+                }
+
+                // select a random trailer from the array of local and remote trailers
+                var ind = Math.floor(Math.random() * trailers.length);
+                var trailer = trailers[ind];
+
+                randomTrailers.push(trailer);
+            }
 
-        }, function (err) {
+            return {"Items": randomTrailers};
```

Comment:
Same here, if we need this information it should just be an API call.

Reply:
Then random trailers should be baked on server side, i.e. all this code should be server side, right?

Reply:
It's how we handle all other filtering/sorting/etc, so it's probably best to handle it server-side, if only for all clients to have the same algorithm.

Reply:
I also find it odd that local and remote trailers are two different endpoints.

Reply:
Any plans for this?

Reply:
@Ullmie02 almost all our endpoints have a sort option that includes `random` as a sort method, are you sure it's not available for the trailer endpoints?